pulsar-broker/src/main/java/org/apache/pulsar/utils/auth/tokens/TokensCliUtils.java

```diff
@@ -0,0 +1,231 @@
+                "--output-public-key" }, description = "File where to write the public key", required = true)
+        String publicKeyFile;
+
+        public void run() throws IOException {
+            KeyPair pair = Keys.keyPairFor(algorithm);
+
+            Files.write(Paths.get(publicKeyFile), pair.getPublic().getEncoded());
+            Files.write(Paths.get(privateKeyFile), pair.getPrivate().getEncoded());
+        }
+    }
+
+    @Parameters(commandDescription = "Create a new token")
+    public static class CommandCreateToken {
+
+        @Parameter(names = { "-s",
+                "--subject" }, description = "Specify the 'subject' or 'principal' associate with this token", required = true)
+        private String subject;
+
+        @Parameter(names = { "-e",
+                "--expiry-time" }, description = "Relative expiry time for the token (eg: 1h, 3d, 10y). (m=minutes) Default: no expiration")
+        private String expiryTime;
+
+        @Parameter(names = { "-pk",
+                "--is-private-key" }, description = "Indicate the signing key is a private key (rather than a symmetric secret key)")
+        private Boolean isPrivateKey = false;
+
+        @Parameter(names = { "-k",
+                "--signing-key" }, description = "Pass the signing key. This can either be: data:, file:, etc..", required = true)
+        private String key;
+
+        public void run() throws Exception {
+            byte[] encodedKey = AuthTokenUtils.readKeyFromUrl(key);
+
+            Key signingKey;
+
+            if (isPrivateKey) {
+                signingKey = AuthTokenUtils.decodePrivateKey(encodedKey);
+            } else {
+                signingKey = AuthTokenUtils.decodeSecretKey(encodedKey);
+            }
+
+            Optional<Date> optExpiryTime = Optional.empty();
+            if (expiryTime != null) {
+                long relativeTimeMillis = TimeUnit.SECONDS
+                        .toMillis(RelativeTimeUtil.parseRelativeTimeInSeconds(expiryTime));
+                optExpiryTime = Optional.of(new Date(System.currentTimeMillis() + relativeTimeMillis));
+            }
+
+            String token = AuthTokenUtils.createToken(signingKey, subject, optExpiryTime);
+            System.out.println(token);
+        }
+    }
+
+    @Parameters(commandDescription = "Show the content of token")
+    public static class CommandShowToken {
+
+        @Parameter(description = "The token string", arity = 1)
+        private java.util.List<String> args;
+
+        @Parameter(names = { "-i",
+                "--stdin" }, description = "Read token from standard input")
+        private Boolean stdin = false;
+
+        @Parameter(names = { "-f",
+                "--token-file" }, description = "Read tokn from a file")
+        private String tokenFile;
+
+        public void run() throws Exception {
+            String token;
+            if (args != null) {
+                token = args.get(0);
+            } else if (stdin) {
```

Comment:
It's pretty standard in unix for '-' to denote stdin. 
So: ```token show -f -```

Reply:
I thought to make it explicit for easier discoverability

Reply:
sure